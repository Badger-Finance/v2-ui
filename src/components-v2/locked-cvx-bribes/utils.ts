import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
