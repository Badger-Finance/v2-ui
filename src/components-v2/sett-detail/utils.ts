import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
