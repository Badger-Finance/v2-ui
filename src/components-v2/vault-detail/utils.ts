import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
