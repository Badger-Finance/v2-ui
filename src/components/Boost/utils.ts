import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
