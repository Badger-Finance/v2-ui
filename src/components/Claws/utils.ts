import BigNumber from 'bignumber.js';
import { INVALID_REASON } from './index';

export interface ValidationParams {
	amount: string | BigNumber;
	maximum: BigNumber;
	minimum?: BigNumber;
}

export function validateAmountBoundaries({
	amount,
	maximum,
	minimum = new BigNumber('0'),
}: ValidationParams): INVALID_REASON | undefined {
	let error: INVALID_REASON | undefined = undefined;

	const input = new BigNumber(amount);
	const amountExceedsBalance = input.gt(maximum);
	const amountIsLessThanMinimum = input.lt(minimum);

	if (amountIsLessThanMinimum) error = INVALID_REASON.UNDER_MINIMUM;
	if (amountExceedsBalance) error = INVALID_REASON.OVER_MAXIMUM;

	console.log({
		input: input.toString(),
		minimum: minimum.toString(),
		maximum: maximum.toString(),
		amountExceedsBalance,
		amountIsLessThanMinimum,
		error,
	});

	return error;
}
