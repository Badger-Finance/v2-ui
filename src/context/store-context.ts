import { createContext } from "react";
import { RootStore } from "../mobx/store";

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;