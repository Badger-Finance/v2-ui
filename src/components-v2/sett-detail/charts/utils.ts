import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
