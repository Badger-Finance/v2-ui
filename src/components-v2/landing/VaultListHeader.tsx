import { Grid, IconButton, makeStyles, Paper, Typography, useMediaQuery, useTheme } from '@material-ui/core';
import clsx from 'clsx';
import { StoreContext } from 'mobx/stores/store-context';
import { observer } from 'mobx-react-lite';
import React, { useContext } from 'react';

import { VaultSortOrder } from '../../mobx/model/ui/vaults-filters';

const useStyles = makeStyles((theme) => ({
	title: {
		display: 'flex',
		textTransform: 'capitalize',
	},
	sortIcon: {
		padding: 4,
	},
	sortInfoIcon: {
		marginLeft: 10,
		width: 20,
	},
	nonSetSort: {
		display: 'none',
		opacity: 0.5,
	},
	nonSetSortMobile: {
		opacity: 0.5,
	},
	columnTitle: {
		flexWrap: 'nowrap',
		'&:hover button:first-of-type': {
			display: 'block',
		},
	},
	tvlColumn: {
		[theme.breakpoints.down('sm')]: {
			display: 'none',
		},
	},
	root: {
		minHeight: 48,
		padding: '10px 42px',
		margin: '15px 0px',
	},
	titlesContainer: {
		paddingLeft: theme.spacing(12),
	},
	spacingItem: {
		width: 106,
		margin: -4,
		[theme.breakpoints.down('md')]: {
			display: 'none',
		},
	},
	sortUp: {
		rotate: '180deg',
	},
}));

const VaultListHeader = observer((): JSX.Element => {
	const classes = useStyles();
	const { vaults } = useContext(StoreContext);
	const { sortOrder } = vaults.vaultsFilters;
	const isMobile = useMediaQuery(useTheme().breakpoints.down('sm'));

	const handleSortByName = (): void => {
		let toggledOrder: VaultSortOrder | undefined;

		switch (sortOrder) {
			case VaultSortOrder.NAME_ASC:
				toggledOrder = undefined;
				break;
			case VaultSortOrder.NAME_DESC:
				toggledOrder = VaultSortOrder.NAME_ASC;
				break;
			default:
				toggledOrder = VaultSortOrder.NAME_DESC;
		}

		vaults.vaultsFilters = { ...vaults.vaultsFilters, sortOrder: toggledOrder };
	};

	const handleSortByApr = (): void => {
		let toggledOrder: VaultSortOrder | undefined;

		switch (sortOrder) {
			case VaultSortOrder.APR_ASC:
				toggledOrder = undefined;
				break;
			case VaultSortOrder.APR_DESC:
				toggledOrder = VaultSortOrder.APR_ASC;
				break;
			default:
				toggledOrder = VaultSortOrder.APR_DESC;
		}

		vaults.vaultsFilters = { ...vaults.vaultsFilters, sortOrder: toggledOrder };
	};

	const handleSortByTvl = (): void => {
		let toggledOrder: VaultSortOrder | undefined;

		switch (sortOrder) {
			case VaultSortOrder.TVL_ASC:
				toggledOrder = undefined;
				break;
			case VaultSortOrder.TVL_DESC:
				toggledOrder = VaultSortOrder.TVL_ASC;
				break;
			default:
				toggledOrder = VaultSortOrder.TVL_DESC;
		}

		vaults.vaultsFilters = { ...vaults.vaultsFilters, sortOrder: toggledOrder };
	};

	const handleSortByBalance = (): void => {
		let toggledOrder: VaultSortOrder | undefined;

		switch (sortOrder) {
			case VaultSortOrder.BALANCE_ASC:
				toggledOrder = undefined;
				break;
			case VaultSortOrder.BALANCE_DESC:
				toggledOrder = VaultSortOrder.BALANCE_ASC;
				break;
			default:
				toggledOrder = VaultSortOrder.BALANCE_DESC;
		}

		vaults.vaultsFilters = { ...vaults.vaultsFilters, sortOrder: toggledOrder };
	};

	return (
		<Grid item container className={classes.root} component={Paper}>
			<Grid container spacing={2}>
				<Grid item xs="auto" className={classes.spacingItem} />
				<Grid item xs lg={4} container alignItems="center" className={clsx(classes.title, classes.columnTitle)}>
					<Typography
						className={classes.title}
						variant="body2"
						color="textSecondary"
						onClick={isMobile ? handleSortByName : undefined}
					>
						Vault
					</Typography>
					{sortOrder !== VaultSortOrder.NAME_ASC && sortOrder !== VaultSortOrder.NAME_DESC && (
						<IconButton
							className={clsx(classes.sortIcon, classes.nonSetSort)}
							onClick={handleSortByName}
							aria-label="sort descending by name"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								alt="sort-icon"
								className={classes.sortIcon}
							/>
						</IconButton>
					)}
					{sortOrder === VaultSortOrder.NAME_DESC && (
						<IconButton
							className={classes.sortIcon}
							onClick={handleSortByName}
							aria-label="sort ascending by name"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								alt="sort-icon"
								className={classes.sortIcon}
							/>
						</IconButton>
					)}
					{sortOrder === VaultSortOrder.NAME_ASC && (
						<IconButton
							className={classes.sortIcon}
							onClick={handleSortByName}
							aria-label="reset sort by name"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								className={clsx(classes.sortUp, classes.sortIcon)}
								alt="sort-icon"
							/>
						</IconButton>
					)}
				</Grid>
				<Grid
					item
					container
					xs
					alignItems="center"
					justifyContent={isMobile ? undefined : 'flex-end'}
					className={clsx(classes.title, classes.columnTitle)}
				>
					<Typography variant="body2" color="textSecondary" onClick={isMobile ? handleSortByApr : undefined}>
						{vaults.vaultsFilters.showAPR ? 'APR' : 'APY'}
					</Typography>
					{sortOrder !== VaultSortOrder.APR_ASC && sortOrder !== VaultSortOrder.APR_DESC && (
						<IconButton
							className={clsx(classes.sortIcon, classes.nonSetSort)}
							onClick={handleSortByApr}
							aria-label="sort descending by APR"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								alt="sort-icon"
								className={classes.sortIcon}
							/>
						</IconButton>
					)}
					{sortOrder === VaultSortOrder.APR_DESC && (
						<IconButton
							className={classes.sortIcon}
							onClick={handleSortByApr}
							aria-label="sort ascending by APR"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								alt="sort-icon"
								className={classes.sortIcon}
							/>
						</IconButton>
					)}
					{sortOrder === VaultSortOrder.APR_ASC && (
						<IconButton
							className={classes.sortIcon}
							onClick={handleSortByApr}
							aria-label="reset sort by APR"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								className={clsx(classes.sortUp, classes.sortIcon)}
								alt="sort-icon"
							/>
						</IconButton>
					)}
				</Grid>
				<Grid
					item
					container
					xs
					alignItems="center"
					justifyContent={isMobile ? undefined : 'flex-end'}
					className={clsx(classes.title, classes.columnTitle)}
				>
					<Typography
						variant="body2"
						color="textSecondary"
						onClick={isMobile ? handleSortByBalance : undefined}
					>
						My Deposits
					</Typography>
					{sortOrder !== VaultSortOrder.BALANCE_ASC && sortOrder !== VaultSortOrder.BALANCE_DESC && (
						<IconButton
							className={clsx(classes.sortIcon, classes.nonSetSort)}
							onClick={handleSortByBalance}
							aria-label="sort descending by balance"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								alt="sort-icon"
								className={classes.sortIcon}
							/>
						</IconButton>
					)}
					{sortOrder === VaultSortOrder.BALANCE_DESC && (
						<IconButton
							className={classes.sortIcon}
							onClick={handleSortByBalance}
							aria-label="sort ascending by balance"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								alt="sort-icon"
								className={classes.sortIcon}
							/>
						</IconButton>
					)}
					{sortOrder === VaultSortOrder.BALANCE_ASC && (
						<IconButton
							className={classes.sortIcon}
							onClick={handleSortByBalance}
							aria-label="reset sort by balance"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								className={clsx(classes.sortUp, classes.sortIcon)}
								alt="sort-icon"
							/>
						</IconButton>
					)}
				</Grid>
				<Grid
					item
					container
					xs
					justifyContent={isMobile ? undefined : 'flex-end'}
					alignItems="center"
					className={clsx(classes.title, classes.columnTitle, classes.tvlColumn)}
				>
					<Typography variant="body2" color="textSecondary">
						TVL
					</Typography>
					{sortOrder !== VaultSortOrder.TVL_ASC && sortOrder !== VaultSortOrder.TVL_DESC && (
						<IconButton
							className={clsx(classes.sortIcon, classes.nonSetSort)}
							onClick={handleSortByTvl}
							aria-label="sort descending by TVL"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								alt="sort-icon"
								className={classes.sortIcon}
							/>
						</IconButton>
					)}
					{sortOrder === VaultSortOrder.TVL_DESC && (
						<IconButton
							className={classes.sortIcon}
							onClick={handleSortByTvl}
							aria-label="sort ascending by TVL"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								alt="sort-icon"
								className={classes.sortIcon}
							/>
						</IconButton>
					)}
					{sortOrder === VaultSortOrder.TVL_ASC && (
						<IconButton
							className={classes.sortIcon}
							onClick={handleSortByTvl}
							aria-label="reset sort by TVL"
						>
							<img
								src="/assets/icons/vaults-sort-icon.svg"
								className={clsx(classes.sortUp, classes.sortIcon)}
								alt="sort-icon"
							/>
						</IconButton>
					)}
				</Grid>
			</Grid>
		</Grid>
	);
});

export default VaultListHeader;
