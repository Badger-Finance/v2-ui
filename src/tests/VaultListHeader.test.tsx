import React from 'react';
import { VaultSortOrder } from '../mobx/model/ui/vaults-filters';
import { customRender, screen, fireEvent } from './Utils';
import VaultListHeader from '../components-v2/landing/VaultListHeader';
import store from '../mobx/RootStore';
import { StoreProvider } from '../mobx/store-context';
import { checkSnapshot } from './utils/snapshots';

type SortTestingOptions = undefined | VaultSortOrder;

describe('Vault List Header', () => {
	it('renders correctly', () => {
		checkSnapshot(
			<VaultListHeader
				title="All Setts"
				helperText="A vault is a smart contract which hold specific tokens. It secures your crypto, while making your money work (e.g. rewards, APR...)"
			/>,
		);
	});

	describe('content sorting', () => {
		describe('by APR', () => {
			test.each([
				[undefined, VaultSortOrder.APR_DESC],
				[VaultSortOrder.APR_DESC, VaultSortOrder.APR_ASC],
				[VaultSortOrder.APR_ASC, undefined],
			])('cycle from %p to %p', (startSortOrder: SortTestingOptions, endSortOrder: SortTestingOptions) => {
				store.uiState.vaultsFilters.sortOrder = startSortOrder;

				let altText: string;

				switch (startSortOrder) {
					case VaultSortOrder.APR_DESC:
						altText = 'sort ascending by APR';
						break;
					case VaultSortOrder.APR_ASC:
						altText = 'reset sort by APR';
						break;
					default:
						altText = 'sort descending by APR';
						break;
				}

				customRender(
					<StoreProvider value={store}>
						<VaultListHeader title="All Setts" />
					</StoreProvider>,
				);

				fireEvent.click(screen.getByLabelText(altText));

				expect(store.uiState.vaultsFilters.sortOrder).toBe(endSortOrder);
			});
		});

		describe('by TVL', () => {
			test.each([
				[undefined, VaultSortOrder.TVL_DESC],
				[VaultSortOrder.TVL_DESC, VaultSortOrder.TVL_ASC],
				[VaultSortOrder.TVL_ASC, undefined],
			])('cycle from %p to %p', (startSortOrder: SortTestingOptions, endSortOrder: SortTestingOptions) => {
				store.uiState.vaultsFilters.sortOrder = startSortOrder;

				let altText: string;

				switch (startSortOrder) {
					case VaultSortOrder.TVL_DESC:
						altText = 'sort ascending by TVL';
						break;
					case VaultSortOrder.TVL_ASC:
						altText = 'reset sort by TVL';
						break;
					default:
						altText = 'sort descending by TVL';
						break;
				}

				customRender(
					<StoreProvider value={store}>
						<VaultListHeader title="All Setts" />
					</StoreProvider>,
				);

				fireEvent.click(screen.getByLabelText(altText));

				expect(store.uiState.vaultsFilters.sortOrder).toBe(endSortOrder);
			});
		});
	});
});
