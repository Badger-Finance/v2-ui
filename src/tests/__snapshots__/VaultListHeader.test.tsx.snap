// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vault List Header renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root-13 makeStyles-root-10 MuiGrid-container-14 MuiGrid-item-15"
  >
    <div
      class="MuiGrid-root-13 makeStyles-titlesContainer-11 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-grid-xs-12-59 MuiGrid-grid-md-9-84 MuiGrid-grid-lg-true-89"
    >
      <div
        class="MuiGrid-root-13 makeStyles-title-1 makeStyles-nameColumn-7 MuiGrid-item-15 MuiGrid-grid-xs-12-59 MuiGrid-grid-md-7-82 MuiGrid-grid-lg-true-89"
      >
        <p
          class="MuiTypography-root-116 makeStyles-title-1 MuiTypography-body2-117 MuiTypography-colorTextSecondary-142"
        >
          All Setts
        </p>
        <img
          alt="List description"
          class="makeStyles-sortInfoIcon-3"
          color="primary"
          src="/assets/icons/vault-sort-info.svg"
          title="A vault is a smart contract which hold specific tokens. It secures your crypto, while making your money work (e.g. rewards, APR...)"
        />
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-1 makeStyles-columnTitle-6 makeStyles-aprColumn-8 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-grid-xs-12-59 MuiGrid-grid-md-true-75"
      >
        <p
          class="MuiTypography-root-116 MuiTypography-body2-117 MuiTypography-colorTextSecondary-142"
        >
          APR
        </p>
        <img
          alt="What is APR?"
          class="makeStyles-sortInfoIcon-3"
          color="primary"
          src="/assets/icons/vault-sort-info.svg"
          title="An annual percentage rate (APR), is the yearly rate earned by staking in each vault."
        />
        <button
          aria-label="sort descending by APR"
          class="MuiButtonBase-root-166 MuiIconButton-root-157 makeStyles-sortIcon-2 makeStyles-nonSetSort-4"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-165"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-169 MuiSvgIcon-colorPrimary-170"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root-178"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-1 makeStyles-columnTitle-6 makeStyles-tvlColumn-9 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-grid-xs-12-59 MuiGrid-grid-md-true-75"
      >
        <p
          class="MuiTypography-root-116 MuiTypography-body2-117 MuiTypography-colorTextSecondary-142"
        >
          TVL
        </p>
        <img
          alt="What is TVL?"
          class="makeStyles-sortInfoIcon-3"
          color="primary"
          src="/assets/icons/vault-sort-info.svg"
          title="Total value locked (TVL) represents the dollar value of all the assets staked in each vault."
        />
        <button
          aria-label="sort descending by TVL"
          class="MuiButtonBase-root-166 MuiIconButton-root-157 makeStyles-sortIcon-2 makeStyles-nonSetSort-4"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-165"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-169 MuiSvgIcon-colorPrimary-170"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root-178"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-1 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-grid-xs-12-59 MuiGrid-grid-md-true-75"
      >
        <p
          class="MuiTypography-root-116 MuiTypography-body2-117 MuiTypography-colorTextSecondary-142"
        >
          MY DEPOSITS
        </p>
        <img
          alt="What represent My Deposits?"
          class="makeStyles-sortInfoIcon-3"
          color="primary"
          src="/assets/icons/vault-sort-info.svg"
          title="It represents the dollar value of all assets, you deposited in each vault."
        />
      </div>
    </div>
  </div>
</div>
`;
